### PR TITLE
including S3 Access Grants Plugin as part of Java SDK Bundle

### PR DESCRIPTION
**Pull request by shiva958**
* As a requirement for Hadoop OpenSource, adding changes to include S3 Access Grants Plugin as part of the Bundle for AWS SDK JAVA V2

## Motivation and Context
Hadoop OpenSource has requested the S3 Access Grants Plugin to be made a part of the SDK bundle for AWS SDK for JAVA V2

## Modifications
* Adding S3 Access Grants plugin as a dependency in aws-sdk-bundle to pull in the dependency when the package is bundled
* Adding version for the plugin in root pom.xml to maintain single place for tracking versions across the package

## Testing
* Packaged aws-sdk-bundle into a JAR file and tested it on a local setup used for the S3 AccessGrants Plugin releases

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)

## Checklist

- [ X] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [X ] Local run of `mvn install` succeeds
- [X ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ X] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Review by amitvc — APPROVED** [review]
LGTM

**Review by millems — COMMENTED** [review]
Can you also add software.amazon.s3.accessgrants:* to the bundle-sdk includeSet? https://github.com/aws/aws-sdk-java-v2/blob/master/bundle-sdk/pom.xml#L70

**Review by millems — APPROVED** [review]

File: aws-sdk-java/pom.xml [hunk 1]
```diff
@@ -728,6 +728,11 @@ Amazon AutoScaling, etc).</description>
             <artifactId>s3control</artifactId>
             <version>${awsjavasdk.version}</version>
         </dependency>
+        <dependency>
+            <groupId>software.amazon.s3.accessgrants</groupId>
+            <artifactId>aws-s3-accessgrants-java-plugin</artifactId>
+            <version>${s3accessgrants.version}</version>
+        </dependency>
```
Comment (dagnir) [inline aws-sdk-java/pom.xml:735]: Can we add this to the `bundle-sdk` POM instead?

File: pom.xml [hunk 2]
```diff
@@ -534,6 +537,9 @@
                              old interfaces, but that should become a test scoped dependency in modules where needed.
                              And the line below removed. -->
                         <ignoredUnusedDeclaredDependency>software.amazon.awssdk:auth</ignoredUnusedDeclaredDependency>
+
+            <!-- Ignoring any unsued delcared dependency warnings across the bundle -->
```
Comment (millems) [inline pom.xml:541]: nit: indentation, s/delcared/declared/

File: pom.xml [hunk 3]
```diff
@@ -177,6 +177,9 @@
         <!-- Reactive Streams version -->
         <reactive-streams.version>1.0.4</reactive-streams.version>
 
+        <!-- S3 Access Grants plugin version -->
+        <s3accessgrants.version>2.0.0</s3accessgrants.version>
```
Comment (millems) [inline pom.xml:181]: Will wait until the patch release to remove the test dependency before merging.

**Review by dagnir — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (shiva958) [reply pom.xml]: Thank You @millems , Fixed it.

File: pom.xml (hunk 3 above)
Comment (shiva958) [reply pom.xml]: Thanks @millems, bumped up the version.

File: pom.xml [hunk 4]
```diff
@@ -534,6 +537,9 @@
                              old interfaces, but that should become a test scoped dependency in modules where needed.
                              And the line below removed. -->
                         <ignoredUnusedDeclaredDependency>software.amazon.awssdk:auth</ignoredUnusedDeclaredDependency>
+
+            <!-- Ignoring any unused declared dependency warnings for access grants plugin across the bundle -->
```
Comment (millems) [inline pom.xml:541]: Nit: indentation

Comment (shiva958) [reply pom.xml]: Thank You. I fixed it.

**Review by millems — APPROVED** [review]

**Pull request merged**